utilities/provider_inventory.py

```
@@ -0,0 +1,201 @@
+        TypeError: If vSphere provider inventory is not VsphereForkliftInventory
+        TimeoutExpiredError: If a VM does not appear in inventory within the timeout
+    """
+    workaround_active = jira_issue_open(INVENTORY_SYNC_WORKAROUND_JIRA) is not False
+    is_vsphere = source_provider.type == Provider.ProviderType.VSPHERE
+    vsphere_inventory: VsphereForkliftInventory | None = None
+
+    if is_vsphere and workaround_active:
+        if not isinstance(source_provider_inventory, VsphereForkliftInventory):
+            raise TypeError(
+                f"vSphere provider requires VsphereForkliftInventory, got {type(source_provider_inventory).__name__}"
+            )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)